File: nlp_sample/count_vectorizer_within_analyzer/src/analyzer.rs
use lindera::tokenizer::{Tokenizer, Token};


pub struct TextAnalyzer {
    tokenizer: Tokenizer
}


impl TextAnalyzer {
    pub fn new() -> Self {
        Self {
            tokenizer: Tokenizer::new().unwrap()
        }
    }
}


impl TextAnalyzer {
    pub fn analyze<'a>(&'a self, text: &'a str) -> Vec<Token> {
        let tokens = match self.tokenizer.tokenize(&text) {
            Ok(tokens) => tokens,
            Err(_e) => Vec::<Token>::new()
        };
        return tokens;
    }
}

File: nlp_sample/example_lda_embedding/analyzer.py
from functools import partial
from typing import List

from sudachipy import dictionary
from sudachipy.tokenizer import Tokenizer


class TextAnalyzer:
    def __init__(self) -> None:
        tokenizer = dictionary.Dictionary().create()
        self._tokenizer = partial(tokenizer.tokenize, mode=Tokenizer.SplitMode.C)

    def analyze(self, text: str) -> List[str]:
        tokens = []
        for token in self._tokenizer(text):
            if token.part_of_speech()[0] == "名詞":
                tokens.append(token.surface())
        return tokens
